lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -265,10 +287,6 @@ public function create_item_permissions_check( $request ) {
 
 		$post_type = get_post_type_object( $this->post_type );
 
-		if ( ! empty( $request['password'] ) && ! current_user_can( $post_type->cap->publish_posts ) ) {
-			return new WP_Error( 'rest_cannot_publish', __( 'Sorry, you are not allowed to create password protected posts in this post type' ), array( 'status' => rest_authorization_required_code() ) );
```

Comment:
Not 100% sure why this special case is needed, do password-protected posts have different capability requirements than regular posts? I'd not have expected to see a custom error message for the password-protected case here.